### ISSUE TITLE
print_linter() for usages of print() where message()/cat() are intended

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#1894]
I see code like this often:

```r
print("Beginning model fitting...")
```

Almost surely this is a mistake -- the `print()` output for character input is worse than `message()`/`cat()` for this purpose:

```r
print("hi")
# [1] "hi"
message("hi")
# hi
```

OTOH, something like `print(iris)` is probably good and being used as intended, so we'll have to come up with a more selective rule than just "`print()`=bad".

I think a good start would be `print(<char literal>)` and `print(foo(...))` for common known-character functions `foo` like `paste()`, `paste0()`, `sprintf()`.
